src/applications/personalization/preferences/containers/SetPreferences.jsx

```diff
@@ -65,7 +62,12 @@ class SetPreferences extends React.Component {
   };
 
   handleSave = () => {
-    this.props.savePreferences(this.props.preferences.dashboard);
+    const { dashboard } = this.props.preferences;
+    if (Object.keys(dashboard).length) {
```

Comment:
Could remove a prop from this component by pushing this decision into the action creator. Few thoughts behind this: 
- there's not really a delete behavior on this for the user to see or use
- it's not really clear when the delete prop would be called just from looking at what the component renders (e.g. if I need to update this and I'm working back from the markup)
- makes it easier for maintainers if they can understand how a component works just from its props  
- this prop could be renamed something the represents what the view is doing (`preferencesChanged`) instead of what the backend needs to do (`savePreferences`)

Reply:
I definitely had the same kind of idea but opted to keep the action creators simple and dumb. But you've convinced me that the component should the dumb one

Reply:
Yeah- I'm not sure what the best justification for that is. Mostly just experience judgment for me- separate business logic from display logic. Usually easier / lower cost to change pure functions too.